feat(setup): setups django with django-rest-framework

wadleo opened the pull request:
## Description
Installs Django with the necessary dependencies for a rest-api
starts a sample test for the user model.

All pull requests should be related to open issues. Indicate the issue(s) bellow and remove this line:
Fixes #1

## How Has This Been Tested?
Black and Flake8 all passed with no warnings/errors
The sample test to valid the inbuilt User and Group models in Django all passed


## Checklist:
<!--- Put an `x` in all the boxes that apply! -->
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [x] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes

wadleo committed:
feat(setup): setups django with django-rest-framework,
starts a test module for the user model.
addresses #1

wadleo committed:
feat(setup): removes an unused webserver gunicorn
also puts a poetry style name for our core module
part of #1

sonegillis reviewed (changes requested):
Great work @wadleo, since it is a project setup feature I suggest,
- You add database setup in the ```settings.py``` like the engine, host, username, db name
- Decouple ```settings.py``` file from key information like the ```SECRET_KEY``` and database host, password, etc. I suggest using the python-decouple package to do that. (As we can see now we have the ```SECRET_KEY``` exposed on git)

gastonche reviewed (dismissed):
Thanks for the changes made here. I don't know about recommendations, but Thanks for updating the README at once.

wadleo committed:
feat(setup): extracts settings to env variables
uses python-decouple to isolate env settings from core app
adds a sample .env file
update the env to reflect the changes
address #1

sonegillis reviewed (approved):
Looks good to me. While I'll like us to change the database engine to MySQL as agreed, I'll approve
